src/Clide.Interfaces/Settings.cs

```diff
@@ -0,0 +1,194 @@
+﻿namespace Clide
+{
+	using Clide.Properties;
+	using System;
+	using System.ComponentModel;
+	using System.Linq;
+	using System.Diagnostics;
+
+	/// <summary>
+	/// Helper base class that can be used to provide transparent loading and saving of settings. 
+	/// </summary>
+	/// <remarks>
+	/// Derived classes typically expose an interface that is exported to the composition container, 
+	/// and declares an importing constructor that receives the settings manager, which is already 
+	/// exported in the environment by the runtime.
+	/// </remarks>
+	/// <example>
+	/// The following is an example of a settings class:
+	/// <code>
+	/// [Settings]
+	/// public class ServerSettings : Settings, IServerSettings
+	/// {
+	///     public ServerSettings(ISettingsManager manager)
+	///         : base(manager)
+	///     {
+	///     }
+	///     
+	///     public string Name { get; set; }
+	///     public int Port { get; set; }
+	/// }
+	/// </code>
+	/// Note how the class specifies what is the exported settings interface 
+	/// for other consuming code. Also, the imported settings manager is passed 
+	/// to the base class which takes care of reading and saving the state as 
+	/// necessary.
+	/// </example>
+	public abstract class Settings : ISettings, INotifyPropertyChanged, ISupportInitialize, ISupportInitializeNotification
+    {
+        private ITracer tracer;
+
+        /// <summary>
+        /// Occurs when initialization of the component is completed.
+        /// </summary>
+        public event EventHandler Initialized = (sender, args) => { };
+
+        /// <summary>
+        /// Occurs when a property value changes.
+        /// </summary>
+        public event PropertyChangedEventHandler PropertyChanged = (sender, args) => { };
+
+        private bool editing;
+        private bool initializing;
+        private ISettingsManager manager;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Settings"/> class.
+        /// </summary>
+        /// <param name="manager">The settings manager that will read and save data for this instance.</param>
+        public Settings(ISettingsManager manager)
```

Comment:
I think that now that we'll be using MEF for these, we might just make the constructor parameterless (makes it easier to just inherit a `Settings` class) and have the `ISettingsManager` as a private/protected `[Import]`ed property. We can then call `Read` on the object when the manager is set, and call `ISupportInitialize.Begin` before reading the values, and `EndInit` afterwards, in case the derived settings class needs to do some post-processing... 

Would simplify the programming model. 

Obviously, for backwards compatibility, I'd leave the (protected, I'd say) ctor overload.